modules/mono/editor/bindings_generator.cpp

```diff
@@ -2731,7 +2747,11 @@ Error BindingsGenerator::_generate_cs_method(const BindingsGenerator::TypeInterf
 
 			if (!p_imethod.is_compat && iarg.default_argument.size()) {
 				if (iarg.def_param_mode != ArgumentInterface::CONSTANT) {
-					arguments_sig += " = null";
+					if (use_span_for_arg) {
+						arguments_sig += " = default";
+					} else {
+						arguments_sig += " = null";
+					}
```

Comment:
I see. In this case, two methods unfortunately have to be skipped in code generation, that are:
```csharp
public Error RequestRaw(
    string url, 
    string[] customHeaders = null, 
    HttpClient.Method method = (HttpClient.Method)(0), 
    byte[] requestDataRaw = null)
```
```csharp
public void PushParagraph(
    HorizontalAlignment alignment, 
    Control.TextDirection baseDirection = (Control.TextDirection)(0), 
    string language = "", 
    TextServer.StructuredTextParser stParser = (TextServer.StructuredTextParser)(0), 
    TextServer.JustificationFlag justificationFlags = (TextServer.JustificationFlag)(163), 
    float[] tabStops = null)
```
Because we cannot have message signature of the following:
```csharp
public Error RequestRaw(
    string url, 
    string[] customHeaders = null, 
    HttpClient.Method method = (HttpClient.Method)(0), 
    ReadOnlySpan<byte> requestDataRaw)
```
```csharp
public void PushParagraph(
    HorizontalAlignment alignment, 
    Control.TextDirection baseDirection = (Control.TextDirection)(0), 
    string language = "", 
    TextServer.StructuredTextParser stParser = (TextServer.StructuredTextParser)(0), 
    TextServer.JustificationFlag justificationFlags = (TextServer.JustificationFlag)(163), 
    ReadOnlySpan<float> tabStops)
```

Reply:
For now, I'm adding an extra check to prevent these span-based APIs that won't compile from being generated.

Reply:
I meant that overloads should not have any optional parameters. Like this:

```csharp
public Error RequestRaw(
    string url,
    string[] customHeaders = null,
    HttpClient.Method method = (HttpClient.Method)(0),
    byte[] requestDataRaw = null)

public Error RequestRaw(
    string url,
    ReadOnlySpan<string> customHeaders,
    HttpClient.Method method,
    ReadOnlySpan<byte> requestDataRaw)
```

Then we don't have to skip those overloads.

Reply:
I see, but I still wish to restore the optional parameters once we have the OverloadResolutionPriorityAttribute.

Reply:
That'd be fine, we can add new default parameter values as long as we don't introduce ambiguity which the attribute should take care of.

It should also be possible to remove the default parameters from the array overloads, I think that shouldn't create ambiguity either because arrays would be implicitly converted to spans but I'd need to check.

However, this last option may be more difficult to implement since not every method has span overloads and we'd need to know before generating the array overload to determine whether to add default parameter values or not.

Reply:
I removed the default parameters for ROS-based APIs (and the guard logic in the binding generator); the PR should be ready once the CI is completed successfully.